src/pages/EditProfilePage.js

```diff
@@ -243,7 +242,7 @@ const EditProfilePage = () => {
                 fullWidth
                 required
                 sx={editProfileClasses.nameField}
-                onChange={(e) => setUsername(e.target.value)}
+                onChange={(e) => user._id === userId && setUsername(e.target.value)}
```

Comment:
Why you don't use other alternatives instead of setting a state for each input component. This will trigger a re-render in whole EditProfilePage. I would recommend use a package like [React Hook Form](https://react-hook-form.com/) to avoid unnecessary renders

Reply:
🤔 well thought. I will study the case and implement it in a new task.